jps-plugin/src/org/jetbrains/jps/incremental/scala/data/CompilationData.scala

```diff
@@ -77,11 +77,17 @@ object CompilationData {
 
       val canonicalSources = sources.map(_.getCanonicalFile)
 
+      val hydraSettings = SettingsManager.getHydraSettings(context.getProjectDescriptor.getProject)
+      val sourcePathOption = if(hydraSettings.isHydraEnabled && hydraSettings.getArtifactPaths.containsKey(scalaVersionFrom(classpath)))
```

Comment:
It would be good to use the code formatting style of the rest of the project. I would break this line differently:
```
val sourcePathOption = 
  if (hydraSettings.isHydraEnabled && hydraSettings.get....
    Seq("-sourcepath", outputGroups.map(_._1).mkString(File.pathSeparator), "cpus", "2")
  else
    Seq.empty
```

You can also select your code and use IntelliJ auto-formatting. A few things: use a space after keywords (`if`), indent the two branches of the `if` relative to the `if` or the `else`...